lab_5_scrapper/scrapper.py

```diff
@@ -13,102 +70,166 @@ def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+        self._validate_config_content()
+        config = self._extract_config_content()
+
+        self._seed_urls = config.seed_urls
+        self._num_articles = config.total_articles
+        self._headers = config.headers
+        self._encoding = config.encoding
+        self._timeout = config.timeout
+        self._should_verify_certificate = config.should_verify_certificate
+        self._headless_mode = config.headless_mode
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            config = json.load(f)
+        return ConfigDTO(**config)
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters
         are not corrupt
         """
-        pass
+        config = self._extract_config_content()
+
+        if not isinstance(config.seed_urls, list) or not config.seed_urls:
+            raise IncorrectSeedURLError('Seed URLs is not a list')
+        for url in config.seed_urls:
+            if not isinstance(url, str):
+                raise IncorrectSeedURLError('Seed URL is not a string')
+            if not re.match(r"https?://w?w?w?.", url):
+                raise IncorrectSeedURLError('Seed URL does not match standard pattern')
+
+        if not isinstance(config.total_articles, int):
+            raise IncorrectNumberOfArticlesError('Total number of articles is not integer')
+
+        if config.total_articles > NUM_ARTICLES_UPPER_LIMIT or config.total_articles < 1:
+            raise NumberOfArticlesOutOfRangeError('Total number of articles is out of range')
+
+        if not isinstance(config.headers, dict):
+            raise IncorrectHeadersError('Headers are not in dictionary form')
+
+        if not isinstance(config.encoding, str):
+            raise IncorrectEncodingError('Encoding is not a string')
+
+        if not isinstance(config.timeout, int):
+            raise IncorrectTimeoutError('Timeout value is not an integer')
+        if not TIMEOUT_LOWER_LIMIT <= config.timeout <= TIMEOUT_UPPER_LIMIT:
```

Comment:
It looks like it may be inclusive: if `timeout` value equals one of the limit values, but does not exceed it, it is acceptable